interface/tasks.py

```diff
@@ -28,28 +28,40 @@ def db_store(question, user, result, ac, wa, job_id, contest, code, lang):
 
 @app.task
 def execute(question, coder, code, lang, contest):
-    language = {"c": "c", "c++": "cpp", "java": "java", "python2": "py", "python3": "py"}
-    ext = language[lang]
-    filename = execute.request.id.__str__() + "." + ext
     try:
-        with open(os.path.join(OUTPATH_DIR, filename), "w+") as file:
-            file.write(code)
-            file.close()
-    except:
-        print("Exception")
-    f = os.path.join(OUTPATH_DIR, filename)
-    question = Question.objects.get(question_code=question['question_code'])
-    user = Coder.objects.get(email=coder['email'])
-    contest = Contest.objects.get(contest_code=contest['contest_code'])
-    testcases = Testcases.objects.filter(question=question)
-    temp_output_file = os.path.join(OUTPATH_DIR, execute.request.id.__str__() + ".txt")
-    ac, wa = 0, 0
-    if (ext == "c"):
+        question = Question.objects.get(question_code=question['question_code'])
+        user = Coder.objects.get(email=coder['email'])
+        testcases = Testcases.objects.filter(question=question)
+        contest = Contest.objects.get(contest_code=contest['contest_code'])
+        ac, wa = 0, 0
+        language = contest.contest_langs.get(name=lang)
+        ext = language.ext
+        filename = execute.request.id.__str__() + "." + ext
+        try:
+            with open(os.path.join(OUTPATH_DIR, filename), "w+") as file:
+                file.write(code)
+                file.close()
+        except:
+            print("Exception")
+        f = os.path.join(OUTPATH_DIR, filename)
+        temp_output_file = os.path.join(OUTPATH_DIR, execute.request.id.__str__() + ".txt")
+        time, mem = 0, 0
         net_res = []
+        if (ext == "c" or ext == "cpp"):
```

Comment:
If you are having problem because of c/c++ being compiled and python/java being interprated. You can use can identified field in the model to differentiate between them.

Reply:
This was not about interpreted vs compiled problem it was basically just that they used the same formatting for their commands so they were grouped together